pgctl/service.py

```diff
@@ -24,13 +24,16 @@
 from .functions import exec_
 from .functions import show_runaway_processes
 from .functions import symlink_if_necessary
+from .functions import terminate_runaway_processes
 from .subprocess import check_call
 from .subprocess import Popen
 
 
-def flock(path):
+def flock(path, force_release=False):
     """attempt to show the user a better message on failure, and handle the race condition"""
     def handle_race(path):
+        if force_release:
+            terminate_runaway_processes(path)
         show_runaway_processes(path)
```

Comment:
I think it'd be better to do this from within show-runaway-processes, after listing them to the user.

Maybe?